content/post/practical-introR.Rmd

```diff
@@ -0,0 +1,697 @@
+
+For importing datasets from excel we can use the library _readxl_ which is linked to tidyverse. However to will still need to load the _readxl_ library as it is not a core tidyverse package.
+
+```{r, eval=FALSE}
+library(readxl)
+dat <- read_excel("PHM-EVD-linelist-2017-10-27.xlsx")
+
+```
+
+	
+```{r include = FALSE}
+library(readxl)
+library(here)
+dat <- readxl::read_xlsx(here("static/data/PHM-EVD-linelist-2017-11-25.xlsx"))
+
+```
+
+
+Here, we will take a look at some of the most used functions from  *tidyverse* 
+
+Here, we will be using a lot the pipe function _%>%_ which is key to use tidyverse and makes programming easier. The pipe function allows emphasise a sequence of actions on an object. 
+
+From the package  *dyplr*, the most common functions are:
+
+__glimpse__: allows a rapid exploration of a dataset
+
+__arrange__: arranges the dataset by the value of a particular variable if numeric, or by alphabetic order if it is a character.
+
+__mutate__: allows generate a new variable
+
+__rename__: allows change a variable's name
+
+__summarise__: reduces the dimension of a dataset 
+
+
+
+
+```{r}
+glimpse(dat)
+
+dat %>% arrange (age)
+dat %>% mutate (fecha_inicio_sintomas = onset)
+
+dat %>% rename (edad = age)
+glimpse(dat)
+
+dat %>% group_by(sex) %>% summarise(number = n())
+
+dat %>% filter(age >14)
+
+select(dat, starts_with("date"))
+select(dat, ends_with("loc"))
+
+slice(dat, 10:15)
+dat[10:15,]
+
+filter(dat, sex == "female", age <= 30)
+
+```
+
+Let's open and explore a dataset that is part of a package
+
+
+```{r, echo = TRUE, include=TRUE}
+# install.packages("outbreaks")
+library(outbreaks)
+measles_dat <- outbreaks::measles_hagelloch_1861
+class(measles_dat)
+head(measles_dat)
+tail(measles_dat)
+
+```
+
+
+
+From the package  *tidyr*, the most common functions are:
+
+__gather__: makes wide data 
+__spred__ : makes long data
+
+
+
+```{r}
+
+people <- tibble(
+  birth_date = as.Date('2000-01-01') + 0:9,
+  name = letters[1:10],
+  Y = rnorm(10, 0, 2),
+  Z = rnorm(10, 0, 4),
+  gender = rep(c('f', 'm'), 5)
+)  %>%
+  mutate(age = as.Date('2019-06-01') -  birth_date)
+
+glimpse(people)
+
+people %>% spread(key = 'age', gender)
+
+
+
+
+```
+
+
+
+
+# ggplot2
+
+GGplot is an implementation of the concept of *grammar of grapphics* that has been implemmented in R by Hadley Wikcham. He explains in his ggplot2 book that "the grammar is a mapping from data to aesthesic atributes (colour, shape, size) of geomethic objects (points, lines, bars)". 
+
+The main components of a ggplot2 plot are:
+- _data frame_ 
+
+- _aesthesic mappings_ this reffers to the indications on how teh data should be mapped (x, y) to color, size, etc
+
+_geoms_ reffers to geometric objects such as points, lines, shapes
+
+_facets_  for condictional plors
+
+_coodinate system_ 
+
+### Basic functions in ggplot
+
+_ggplo()_ is the core function in ggplot2. The basic argument of the function is the dataframe we want to plot _(data)_.
```

Comment:
for functions, it's preferred to use the code font: `ggplot()`

Reply:
Got it